### ISSUE TITLE
Consumption Plan Linux - 0 functions loaded for .NET Core 3.1 Functions project

### ISSUE THREAD

**Issue by sbosman84** [Azure/azure-functions-host#6884]
We've tested this behavior with a new clean .NET Core 3.1 Azure Functions project containing only a TimerTrigger function which executes every 5 minutes. When we deploy this functions to a Azure Function based on a Linux AppService plan the function is loaded correctly and the timer trigger is executed succesful.

But when we deploy the same function to a Linux Consumption Plan we don't see the functions being loaded. And also a App Insights messages states that 0 functions have been loaded:
![image](https://user-images.githubusercontent.com/15688831/98795210-40c72d00-240a-11eb-8ea0-fc9226b60970.png)

#### Investigative information

- Timestamp: 11-11-2020 09:53:44 AM (CEST)
- Function App version: 3.0.14916.0
- Function App name: function-linux
- Function name(s) (as appropriate): TimerTrigger
- Invocation ID: -
- Region: West Europe

#### Repro steps

1. Create new Azure Functions project in Visual Studio (.NET Core 3.1)
2. Add timer trigger function to the project
3. Create Function App based on Consumption Plan, Operating System: Linux
4. Deploy project
5. Look at functions in created function app and no function will show also AppInsights logging indicates: 0 functions loaded

#### Expected behavior

At step 5 I expect that the TimeTrigger function is loaded correctly

#### Actual behavior

Functions in created function app are not loading and AppInsights log indicated that 0 functions have been loaded

#### Known workarounds

Provide a description of any known workarounds.

Deploy the function on a Function App based on a AppService plan for Linux and it runs fine:
![image](https://user-images.githubusercontent.com/15688831/98795929-1de94880-240b-11eb-887b-41088861003a.png)

#### Related information 
Code of the function;
<summary>Source</summary>

```csharp
        [FunctionName("TimerTrigger")]
        public static void Run([TimerTrigger("0 */5 * * * *", RunOnStartup = true)]TimerInfo myTimer, ILogger log)
        {
            log.LogInformation("It's working now!");            
        }
```

**Comment by v-anvari** [comment]
Hi @sbosman84, Functions cannot be edited in linux - consumption plan in portal, this feature is not supported yet, which is the reason you see "0 functions loaded".

Try to create functions using Powershell [create-first-function-cli](https://docs.microsoft.com/en-us/azure/azure-functions/create-first-function-cli-python?tabs=azure-powershell%2Cbash%2Cbrowser#configure-your-local-environment)

**Related information:**
[Azure Function App on Linux consumption plan](https://github.com/Azure/azure-functions-host/issues/6857#issuecomment-724751105)

You can find the discussion in [MicrosoftDocs/azure-docs/issues/60172- Editing functions in the Azure portal is not supported for Linux Consumption Function Apps](https://github.com/MicrosoftDocs/azure-docs/issues/60172).

**Comment by anthonychu** [comment]
@vijayrkn @stefanushinardi Can you please confirm that we don’t support publishing to Linux Consumption from VS?

**Comment by vijayrkn** [comment]
@anthonychu - We do support creating and publishing to Functions Linux (consumption) from VS
![image](https://user-images.githubusercontent.com/3981619/99554536-27773100-2974-11eb-9ffb-c1792476164e.png)

**Comment by v-anvari** [comment]
Hi @anthonychu , @vijayrkn , We have the option in VS to deploy Function App, but the functions aren't getting deployed to the function app. I have tried to publish the functions, but VS doesn't let me publish it. Also, tried different workarounds to publish the function through VS, but they never show up in portal. One of the team members confirmed that functions is not supported in Linux consumption

**Comment by sbosman84** [comment]
@v-anvari So if I understood it correctly, It is just publishing from Visual Studio which is not supported and results in the functions not getting deployed in a linux consumption plan. But a publish via for example Azure Pipelines should work?

**Comment by vijayrkn** [comment]
I don't see any reason why a publish from VS wouldn't work if it works from Azure DevOps. Do you get any error while publishing from VS? Is there a sample that we can look at?

**Comment by anthonychu** [comment]
Reopening as I think it’s important to investigate and fix any issues that are found.

**Comment by sbosman84** [comment]
> I don't see any reason why a publish from VS wouldn't work if it works from Azure DevOps. Do you get any error while publishing from VS? Is there a sample that we can look at?

It is just a default timer trigger function. Can create a github project for you if it is needed? We don't get any error and the publish is successful. Strange thing is when we deploy the same function to a Function running on a Linux app service plan. The deployment works fine.

**Comment by vijayrkn** [comment]
@sbosman84 - if the same app without any changes works in Linux (App Service plan) but not in Linux (consumption), that feels more like an issue on the service side. If you can share a GitHub repo with the project, I can give that a try as well to see if I can track down where the issue is.

**Comment by sbosman84** [comment]
@vijayrkn I've pushed the project, I used to test, to the following repo: https://github.com/sbosman84/personal. I've created the Azure Function running on a Linux Consumption Plan via the portal and downloaded the publish profile which I imported in the Visual Studio publish wizard. 

When I deploy this version to a Linux Consumption Plan it doesn't show any function:
![image](https://user-images.githubusercontent.com/15688831/99592281-4f0bdf00-29f0-11eb-8cf6-a5f9df8063f7.png)

When I deploy the same function to a Linux App Service Plan it does show the function:
![image](https://user-images.githubusercontent.com/15688831/99592336-60ed8200-29f0-11eb-9d65-72e93f5b4883.png)

**Comment by vijayrkn** [comment]
@sbosman84 - After you right clicked -> Publish, did you import the profile or select the profile from the Azure selection UI? 

If you imported, can you try going through the wizard flow Azure -> Functions(Linux) -> Pick the instance and then create the profile.

**Comment by sbosman84** [comment]
@vijayrkn Yes, I've imported the profile. Just tried the wizard flow Azure -> Functions(Linux) -> function-linux but got an error while publishing:
![image](https://user-images.githubusercontent.com/15688831/99593330-cdb54c00-29f1-11eb-9320-9801bbca2c48.png)

Also got the same error when publishing via the same route to the App Service Plan function:
![image](https://user-images.githubusercontent.com/15688831/99593358-d7d74a80-29f1-11eb-92f3-32cdd10d9e25.png)

**Comment by vijayrkn** [comment]
ok that explains! There is a known issue that Azure functions team is fixing. As a work-around, can you add this app setting "WEBSITE_WEBDEPLOY_USE_SCM=true", delete the profile and go through the selection UI again. That should fix this error.

**Comment by vijayrkn** [comment]
Oops dangerous typo. The value should be set to true instead of false. I have fixed the above comment. Sorry about that.

**Comment by vijayrkn** [comment]
With the above fix, I was able to publish and see the function in linux consumption 
![image](https://user-images.githubusercontent.com/3981619/99594659-53aca500-29a8-11eb-9148-d86423defcf6.png)

**Comment by sbosman84** [comment]
Yes, I've the same behavior! Via a zip deploy (and the app setting) I'm able to deploy the function on a consumption plan as well. 

So the issue lies with the publish via a web deploy then? But at least I've a workaround with the publish via zip deploy. Thanks for the support!

**Comment by vijayrkn** [comment]
Thanks for the confirmation. The functions team is already aware of the issue and the deployment should happen soon. 
/cc @anthonychu

**Comment by FelipeAlvesLima** [comment]
?
